tests/emukit/core/test_user_function.py

```diff
@@ -0,0 +1,70 @@
+import pytest
+import numpy as np
+from numpy.testing import assert_array_equal
+
+from emukit.core.loop import UserFunctionWrapper, UserFunctionResult
+
+
+def test_user_function_wrapper_evaluation_no_cost():
+    function = lambda x: 2 * x
+    function_input = np.array([[1], [2], [3]])
+    ufw = UserFunctionWrapper(function)
+
+    output = ufw.evaluate(function_input)
+
+    assert len(output) == function_input.shape[0]
+    for i, record in enumerate(output):
+        assert_array_equal(output[i].X, function_input[i])
+        assert_array_equal(output[i].Y, function(function_input[i]))
+        assert output[i].cost is None
+
+
+def test_user_function_wrapper_evaluation_with_cost():
+    function = lambda x: (2 * x, np.array([[1]] * x.shape[0]))
+    function_input = np.array([[1], [2], [3]])
+    ufw = UserFunctionWrapper(function)
+
+    output = ufw.evaluate(function_input)
+
+    assert len(output) == function_input.shape[0]
+    for i, record in enumerate(output):
+        assert output[i].X == function_input[i]
+        assert output[i].Y == function(function_input[i])[0]
```

Comment:
Is there a reason you use `assert_array_equal` in the test function above but not here?